src/Tagged.elm

```diff
@@ -1,94 +1,90 @@
 module Tagged exposing (..)
 
-{-|
+{-| A module that allows you to "tag" a value.
 
-A module that allows you to "tag" a value.
+@docs Tagged, tag, retag, untag, map, andMap, map2, andThen, extend
 
-@docs Tagged, tag, retag, untag, map, ap, map2, bind, extend
 -}
 
-{-|
-A Type that pairs a `value` with a `tag`.
+
+{-| A Type that pairs a `value` with a `tag`.
 
 The `tag` is ignored at runtime as evidenced by the only value constructor:
 
-```elm
-Tagged : value -> Tagged tag value
-```
+    Tagged : value -> Tagged tag value
+
 -}
 type Tagged tag value
-  = Tagged value
+    = Tagged value
 
-{-|
-An alias for the `Tagged` value constructor.
+
+{-| An alias for the `Tagged` value constructor.
 -}
 tag : value -> Tagged tag value
 tag =
-  Tagged
+    Tagged
+
 
-{-|
-Useful for composing functions together infix:
+{-| Useful for composing functions together in a pipeline:
+
+    foo =
+        Array.set |> map index |> andMap value |> andMap arr
```

Comment:
Should the pipeline actually be this?

```elm
index |> map Array.set |> andMap value |> andMap arr
```

I find that
```elm
value1
  |> map function
  |> andMap value2
``` 

is harder to read than

```elm
succeed function
  |> andMap value1
  |> andMap value2
```

The combination of `map` and `andMap` reads nicely in infix form but not so much with pipelines.

I suggest the following for the docs:

```elm
foo =
  tagged Array.set |> andMap index |> andMap value |> andMap arr
```

This keeps the function, arg1, arg2, ... order. It also has a nicer symmetry where `tagged` is used for the combining function and an `andMap` is used to apply each argument.

Reply:
Good catch, and fair point. Since this is the doc for `map`, though, it would be silly not having `map` in there. I'll change this one to something silly like `map String.toUpper`, and redo the one for `andMap`.

Reply:
LOL, good point. For some reason I thought this was the doc for `andMap` 😛